src/Contexts/PostsContext.jsx

```diff
@@ -19,57 +19,21 @@ export const PostsProvider = ({ children }) => {
 
   useEffect(() => {
     const getPosts = () => {
-      api
-        .get("/posts?_start=0&_limit=20")
-        .then((res) => {
-          setPosts(res.data);
-        })
-        .catch((err) => {
-          console.log("Failed to load 0~20 posts", err);
-        })
-        .then(
-          api
-            .get("/posts?_start=20&_limit=20")
-            .then((res) => {
+      api.get("/posts?_start=0&_limit=20").then((res) => {
+        setPosts(res.data);
+        api.get("/posts?_start=20&_limit=20").then((res) => {
+            setPosts((posts) => [...posts, ...res.data]);
+            api.get("/posts?_start=40&_limit=20").then((res) => {
               setPosts((posts) => [...posts, ...res.data]);
-            })
-            .catch((err) => {
-              console.log("Failed to load 20~40 posts", err);
-            })
-        )
-        .then(
-          api
-            .get("/posts?_start=40&_limit=20")
-            .then((res) => {
-              setPosts((posts) => [...posts, ...res.data]);
-            })
-            .catch((err) => {
-              console.log("Failed to load posts 40~60", err);
-            })
-        )
-        .then(
-          api
-            .get("/posts?_start=60&_limit=20")
-            .then((res) => {
-              setPosts((posts) => [...posts, ...res.data]);
-            })
-            .catch((err) => {
-              console.log("Failed to load posts 60~80", err);
-            })
-        )
-        .then(
-          api
-            .get("/posts?_start=80&_limit=15")
-            .then((res) => {
-              setPosts((posts) => [...posts, ...res.data]);
-            })
-            .catch((err) => {
-              console.log("Failed to load posts 80~95", err);
-            })
-        )
-        .finally(() => {
-          setIsLoading(false);
-        });
+              api.get("/posts?_start=60&_limit=20").then((res) => {
+                setPosts((posts) => [...posts, ...res.data]);
+                api.get("/posts?_start=80&_limit=15").then((res) => {
+                  setPosts((posts) => [...posts, ...res.data]);
+                }).finally(() => {setIsLoading(false)});
```

Comment:
Aqui acho que falta um `catch` antes do `finally`:

```suggestion
                })
                .catch(err => {console.log(err.toJSON());})
                .finally(() => {setIsLoading(false);});
 ```